pkg/internal/ebpf/tracer_linux.go

```diff
@@ -34,10 +37,38 @@ type instrumenter struct {
 	modules   map[uint64]struct{}
 }
 
+func resolveInternalMaps(spec *ebpf.CollectionSpec) (*ebpf.CollectionOptions, error) {
+	collOpts := ebpf.CollectionOptions{MapReplacements: map[string]*ebpf.Map{}}
+
+	for k, v := range spec.Maps {
+		if v.Pinning != PinInternal {
+			continue
+		}
+
+		v.Pinning = ebpf.PinNone
+		internalMap := internalMaps[k]
+
+		var err error
+
+		if internalMap == nil {
+			internalMap, err = ebpf.NewMap(v)
+
+			if err != nil {
+				return nil, fmt.Errorf("failed to load shared map: %w", err)
+			}
+
+			internalMaps[k] = internalMap
+		}
+
+		collOpts.MapReplacements[k] = internalMap
```

Comment:
`cilium/ebpf` will clone `internalMap` when processing `MapReplacements`, meaning both clones can be safely closed irrespective of each other (clones still share the underlying map object though, it's just the underlying file descriptor that is dup'd)